### ISSUE TITLE
build from source is not working with go1.19, go.mod & docu needs update

### ISSUE THREAD

**Issue by gen2thomas** [gohugoio/hugo#11864]
### What version of Hugo are you using (`hugo version`)?

Try to build the latest version (0.121.2), made with go 1.21.5.

### Does this issue reproduce with the latest release?

yes

### go Version

```
go version
go version go1.19.13 linux/amd64
```

### Description

run build for installation according to [documentation](https://gohugo.io/installation/linux/#build-from-source):

"Install Go version 1.19 or later"

```
CGO_ENABLED=1 go install -tags extended github.com/gohugoio/hugo@latest
# github.com/getkin/kin-openapi/openapi3
/home/user/go/pkg/mod/github.com/getkin/kin-openapi@v0.122.0/openapi3/schema_pattern.go:27:32: compiledPatterns.CompareAndSwap undefined (type sync.Map has no field or method CompareAndSwap)
note: module requires Go 1.20
```

Since [go 1.21](https://go.dev/doc/modules/gomod-ref) the go version directive should (citation):

- The go line declares a required minimum version of Go to use with this module.
- The go line must be greater than or equal to the go line of **all dependencies**.
- The go command no longer attempts to maintain compatibility with the previous older version of Go.
- The go command is more careful about keeping checksums of go.mod files in the go.sum file. 

I will try to prepare a PR to fix this in go.mod and also for the documentation site afterwards.

**Comment by gen2thomas** [comment]
I thought the changes for documentation are trivial, but there are many findings for "go 1.19" and additionally an issue for [Changes required for v0.122.0](https://github.com/gohugoio/hugoDocs/issues/2385) already exist. 

@jmooring , how should I proceed?

**Comment by jmooring** [comment]
Yeah, this changed with https://github.com/gohugoio/hugo/commit/f31375d4ccaf6dfffc7c6c21d064034508f6b524, where we bumped [getkin/kin-openapi](https://github.com/getkin/kin-openapi/i) from 0.118.0 to 0.120.0 (see https://github.com/getkin/kin-openapi/issues/836 for details).

I've updated the documentation site (https://github.com/gohugoio/hugoDocs/pull/2387) and the README in this repository.

**Comment by gen2thomas** [comment]
@jmooring thanks for your fast response and fix for the docu part. But the go.mod part is still open. I like to prepare a PR and link this issue. So please can you re-open the issue.

Maybe it was my fault to add an issue for 2 different parts - sorry for that.

**Comment by cmahnke** [comment]
Ist this still relevant with #11865 being closed and #12012 being merged?
